### DISCUSSION TITLE
Automatic type conversion, instead of throwing an error of the wrong data type.

### DISCUSSION THREAD

**Discussion by Fragers** [pydantic/pydantic#3943] [Question] [answered]
Faced with an unusual bug. Such code will automatically cast int to the float type, instead of throwing an error of the wrong type. Is this the right behavior?

```python
class ParsedResult(BaseModel):
    bbox: List[float]
```

**Comment by samuelcolvin** [comment, marked as answer]
Yes this is (currently) the intended behaviour.

In future we might have a consultation about changing this.

In the meantime you can use a validator to raise an error.
